src/controllers/companiesController.js

```diff
@@ -81,8 +81,13 @@ const updateCompany = async (req, res) => {
 
 const deleteCompany = async (req, res) => {
   try {
+    const buildingFromCompany = await models.Building.find({ company: req.params.id });
```

Comment:
What happens if you have a company but it's not used by a building? The `find` method returns an empty array and, in the next line when you do `if (buildingFromCompany)` this will be true because you have an empty array and it's not null.
So, the best way to do this is using `findOne` instead of `find`. `findOne` will return the first building that have the company with the provided id. If there is not a building with that company id, the result will be `null`.
```suggestion
    const buildingFromCompany = await models.Building.findOne({ company: req.params.id });
```